### PR TITLE
Add some unit tests for some target / vector corner cases

### PR DESCRIPTION
**Pull request by mwachs5**
Unit tests for #2778 and some other interesting cases

### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: src/test/scala/chiselTests/LiteralToTargetSpec.scala [hunk 1]
```diff
@@ -27,4 +27,64 @@ class LiteralToTargetSpec extends AnyFreeSpec with Matchers {
       ChiselStage.elaborate(new Foo)
     } should have).message("Illegal component name: UInt<1>(\"h01\") (note: literals are illegal)")
   }
+
+  "Vectors which are indexed by literal values should successfully be converted to Reference Target" in {
```
Comment (mwachs5) [inline src/test/scala/chiselTests/LiteralToTargetSpec.scala:31]: or maybe this is just another case of the next one and shouldn't be expected to work either. We probably just need a better error message for "you're trying to annotate the result of a vector lookup, if you want to do that assign it to a temporary"
